packages/api/src/external/ehr/athenahealth/command/get-patient.ts

```diff
@@ -0,0 +1,129 @@
+import { PatientResource } from "@metriport/shared/interface/external/athenahealth/patient";
+import { normalizeDate, normalizeGender } from "@metriport/shared";
+import { executeAsynchronously } from "@metriport/core/util/concurrency";
+import { out } from "@metriport/core/util/log";
+import { capture } from "@metriport/core/util/notifications";
+import { errorToString } from "@metriport/shared";
+import { Patient, PatientDemoData } from "@metriport/core/domain/patient";
+import { getPatient as getAthenaPatient } from "@metriport/core/external/athenahealth/get-patient";
+import { EhrSources } from "../../shared";
+import {
+  getPatientOrFail as getMetriportPatientOrFail,
+  getPatientByDemo as getMetriportPatientByDemo,
+} from "../../../../command/medical/patient/get-patient";
+import { getPatientMappingId, createPatientMapping } from "../../../../command/mapping/patient";
+import { Config } from "../../../../shared/config";
+import { createMetriportAddresses, createMetriportContacts } from "../shared";
+
+const athenaUrl = Config.getAthenaHealthUrl();
+
+export async function getPatient({
+  accessToken,
+  cxId,
+  athenaPatientId,
+}: {
+  accessToken: string;
+  cxId: string;
+  athenaPatientId: string;
+}): Promise<Patient | undefined> {
+  const { log } = out(`AthenaHealth getPatient - cxId ${cxId} athenaPatientId ${athenaPatientId}`);
+  const existingPatientId = await getPatientMappingId({
+    externalId: athenaPatientId,
+    source: EhrSources.ATHENA,
+  });
+  if (existingPatientId) {
+    return await getMetriportPatientOrFail({ cxId, id: existingPatientId });
+  }
+  if (!athenaUrl) throw new Error("Athenahealth url not defined");
+  const athenaPatient = await getAthenaPatient({
+    accessToken,
+    baseUrl: athenaUrl,
+    patientId: athenaPatientId,
+  });
+  if (!athenaPatient) return undefined;
+  if (athenaPatient.name.length === 0) {
+    throw new Error("AthenaHealth patient missing at lesat one name");
+  }
+  if (athenaPatient.address.length === 0) {
+    throw new Error("AthenaHealth patient missing at lesat one address");
+  }
+
+  const patientDemos = createMetriportPatientDemos(athenaPatient);
+  const patients: Patient[] = [];
+  const getPatientByDemoWrapperErrors: string[] = [];
+
+  async function getPatientByDemoWrapper({
+    cxId,
+    demo,
+    patients,
+    errors,
+    log,
+  }: {
+    cxId: string;
+    demo: PatientDemoData;
+    patients: Patient[];
+    errors: string[];
+    log: typeof console.log;
```

Comment:
Personally I just think the code holds up better overtime without relying on scoping for variables -- by passing them explicitly the function is self-contained. Similar to our discussion around awaits inside functions vs. call them when the function is invoked. It's a style thing that I think makes the code just a little more robust (and obvious).

Reply:
Especially for variables that have common names (e.g. patients).

Reply:
I see your point, but then might as well move the code to its own function and make this one simpler/leaner. Thoughts?

Reply:
For the ones I've introduced, I agree - done. But even a *small* inline function should have it's variables explicitly defined (and not inherited from the scope) IMO -- as variables change and functions expand with multiple devs, being explicit makes it clear what the in-line function expects and it's internal data structures.

Reply:
We can chat about this on a tech-talk, but the only reason to create an inline function IMO is to make use of scope constants. I totally agree w/ being explicit and avoid things that can break in the future (that's why I'm so repetitive to use constants over variables 😄), but I also see how a small inline function that uses consts from the parent function can be fine (using constants and strong types prob saves us from any issues we could have in that situation).

Reply:
Yea, generally I agree what inline functions that inherit scope are fine when they're crystal clear and obvious -- otherwise shouldn't be inline so that the scope element is a none issue 👍